drivers/storage/portworx/deployment.go

```diff
@@ -877,6 +869,45 @@ func (t *template) loadKvdbAuth() map[string]string {
 	return t.kvdb
 }
 
+func (t *template) csiBasePath() string {
+	return path.Join("/var/lib/kubelet/plugins", t.csiVersions.driverName)
+}
+
+func extractPXVersion(cluster *corev1alpha1.StorageCluster) *version.Version {
+	var (
+		err       error
+		pxVersion *version.Version
+	)
+
+	pxImage := cluster.Spec.Image
+	for _, env := range cluster.Spec.Env {
+		if env.Name == envKeyPXImage {
+			pxImage = env.Value
+			break
+		}
+	}
+
+	parts := strings.Split(pxImage, ":")
+	if len(parts) >= 2 {
+		pxVersionStr := parts[len(parts)-1]
+		pxVersion, err = version.NewSemver(pxVersionStr)
+		if err != nil {
+			logrus.Warnf("Invalid PX version %s extracted from image name: %v", pxVersionStr, err)
+			if pxVersionStr, exists := cluster.Annotations[annotationPXVersion]; exists {
+				pxVersion, err = version.NewSemver(pxVersionStr)
```

Comment:
Nice, this is much better than the installer version. We need to use the oci-monitor there and the px version is inferred from that, which isn't always accurate.